File: server/tests/fixtures/alpha_rule_triggers/csharp/arch_entry_point.cs
// Should trigger: arch.entry_point
using Microsoft.AspNetCore.Mvc;

namespace MyApp.Controllers
{
    // ASP.NET Core controller - HTTP entry points
    [ApiController]
    [Route("api/[controller]")]
    public class UsersController : ControllerBase
    {
        [HttpGet]
        public IActionResult GetUsers()
        {
            return Ok(new { users = new string[0] });
        }

        [HttpPost]
        public IActionResult CreateUser([FromBody] object body)
        {
            return Ok(new { created = true });
        }
    }
    
    // Main entry point
    public class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Starting application...");
        }
    }
}

File: server/tests/fixtures/alpha_rule_triggers/python/arch_entry_point.py
# Should trigger: arch.entry_point
# Flask HTTP endpoint
from flask import Flask

app = Flask(__name__)

@app.route('/api/users')
def get_users():
    """HTTP entry point - GET /api/users"""
    return {"users": []}

# Main function entry point
def main():
    """Application main entry point"""
    print("Starting application...")
    app.run()

if __name__ == "__main__":
    main()
